app/models/user.rb

```diff
@@ -4,6 +4,7 @@ class User < ApplicationRecord
   validates :password_digest, confirmation: true
   enum role: %w(regular merchant admin)
   # validates :merchant_id, numericality: {allow_blank: true}
+  validates :merchant, presence: true, if: :merchant? 
```

Comment:
Nice! I'd like to have @d-atkins double check me on this one, because I haven't created logic like this previously.

Reply:
David confirmed that this should be good to go!